recipes/config.rb

```diff
@@ -58,6 +58,9 @@
 service_action = node['filebeat']['disable_service'] ? [:disable, :stop] : [:enable, :nothing]
 
 service 'filebeat' do
+  provider Chef::Provider::Service::Solaris if node['platform_family'] == 'solaris2'
+  retries 30
```

Comment:
I think `retries` and `retry_delay` should be configurable. This way you can modify it according to your environment. I will go ahead with merge and will change it to defaults.